### PR TITLE
[REVERTED] Port MYFACES-4606

### LINKED ISSUE #26221: Port MYFACES-4606 (Issuing Element Not Found in Request Parameter Map for Ajax Requests) to Liberty
**Describe the bug**  
Take the following example, which requires a particular button to be pressed for some input.
```
<h:inputText required="#{not empty param[someButton.clientId]} ... />

<h:commandButton id="someButton" value="Submit" binding="#{someButton}">
    <f:ajax event="click" ... />
</h:commandButton>

```
When 'someButton' is pressed, the JSF Javascript logic should include the button's id in the request parameter map during the request (due to the binding attribute). 
If the issuing element is not sent as part of the request, then it can cause the required check to be evaluated improperly. 
This previously worked in MyFaces 2.0, but not in MyFaces 2.2 (or higher).
_Note that only ajax requests were broken._

**Steps to Reproduce**  
See code above. 

**Expected behavior**  
The id of the pressed button should be included in the request parameters. 

**Diagnostic information:**  
 - OpenLiberty Version: All
 - Affected feature(s) jsf-2.2, jsf-2.3, faces-3.0, faces-4.0
 - Java Version: n/a
 - server.xml configuration: n/a

**Additional context**  
https://issues.apache.org/jira/browse/MYFACES-4606
https://stackoverflow.com/questions/14729982/validate-input-as-required-only-if-certain-command-button-is-pressed/14730658#14730658

### PR DESCRIPTION
**Pull request by volosied**
fixes #26221

### CODE REVIEW

**Comment by posmith** [comment]
L2 message review not required

**Comment by volosied** [comment]
I'll remove the overlays for the message files.

**Comment by dmuelle** [comment]
Hi @volosied -are there new messages here that need ID review? The Libbybot is flagging a bunch of files but can't tell if any of them actually changed.

If not, you can add the `ID review not needed` label to bypass review. Thanks

**Comment by volosied** [comment]
#build

**Comment by volosied** [comment]
#build

**Comment by volosied** [comment]
#build

**Comment by volosied** [comment]
#build

File: dev/com.ibm.ws.jsf.2.2_fat/publish/servers/jsf22APARSeleniumServer/bootstrap.properties [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+# Copyright (c) 2023 IBM Corporation and others.
+# All rights reserved. This program and the accompanying materials
+# are made available under the terms of the Eclipse Public License 2.0
+# which accompanies this distribution, and is available at
+# http://www.eclipse.org/legal/epl-2.0/
+# 
+# SPDX-License-Identifier: EPL-2.0
+#
+# Contributors:
+#     IBM Corporation - initial API and implementation
```
Comment (pnicolucci) [inline dev/com.ibm.ws.jsf.2.2_fat/publish/servers/jsf22APARSeleniumServer/bootstrap.properties]: remove the contributors

File: dev/com.ibm.ws.jsf.2.2_fat/publish/servers/jsf22APARSeleniumServer/server.xml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+ * Copyright (c) 2023 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-2.0/
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
```
Comment (pnicolucci) [inline dev/com.ibm.ws.jsf.2.2_fat/publish/servers/jsf22APARSeleniumServer/server.xml]: remove contributors

File: dev/com.ibm.ws.jsf.2.2_fat/test-applications/PH55398.war/resources/WEB-INF/web.xml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+ * Copyright (c) 2023 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-2.0/
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
```
Comment (pnicolucci) [inline dev/com.ibm.ws.jsf.2.2_fat/test-applications/PH55398.war/resources/WEB-INF/web.xml]: remove contributors

File: dev/com.ibm.ws.jsf.2.2_fat/test-applications/PH55398.war/resources/index.xhtml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
+          "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
+<html xmlns="http://www.w3.org/1999/xhtml" xmlns:h="http://java.sun.com/jsf/html"
+      xmlns:f="http://java.sun.com/jsf/core"
+      xmlns:c="http://java.sun.com/jsp/jstl/core" >
+  <head>
+    <meta http-equiv="Content-Type" content="text/html; charset=iso-8859-1"/>
+    <title>Test PH55398</title>
+  </head>
+<body>
```
Comment (pnicolucci) [inline dev/com.ibm.ws.jsf.2.2_fat/test-applications/PH55398.war/resources/index.xhtml]: Let's clean up this formatting

**Review by pnicolucci — APPROVED** [review]

**Comment by volosied** [comment]
#libby

**Pull request merged**